src/unistd.rs

```diff
@@ -438,37 +452,43 @@ pub fn dup(oldfd: RawFd) -> Result<RawFd> {
 /// specified fd instead of allocating a new one.  See the man pages for more
 /// detail on the exact behavior of this function.
 #[inline]
-pub fn dup2(oldfd: RawFd, newfd: RawFd) -> Result<RawFd> {
-    let res = unsafe { libc::dup2(oldfd, newfd) };
+pub fn dup2<Fd: AsFd>(oldfd: &Fd, newfd: &mut OwnedFd) -> Result<()> {
+    let res =
+        unsafe { libc::dup2(oldfd.as_fd().as_raw_fd(), newfd.as_raw_fd()) };
 
-    Errno::result(res)
+    Errno::result(res).map(drop)
 }
 
 /// Create a new copy of the specified file descriptor using the specified fd
 /// and flags (see [dup(2)](https://man7.org/linux/man-pages/man2/dup.2.html)).
 ///
 /// This function behaves similar to `dup2()` but allows for flags to be
 /// specified.
-pub fn dup3(oldfd: RawFd, newfd: RawFd, flags: OFlag) -> Result<RawFd> {
```

Comment:
Sigh, we still need to get rid of this function.  It isn't Nix's place to be a C library.

Reply:
Yeah, I tried having it use libc::dup3, but that's only available on certain platforms. Should I cfg it to just those platforms?

Reply:
Let's do it in a separate PR.